src/live_restore/live_restore_fs.c

```diff
@@ -597,72 +480,51 @@ typedef enum { NONE, FULL, PARTIAL } WT_LIVE_RESTORE_SERVICE_STATE;
  */
 static WT_LIVE_RESTORE_SERVICE_STATE
 __live_restore_can_service_read(WTI_LIVE_RESTORE_FILE_HANDLE *lr_fh, WT_SESSION_IMPL *session,
-  wt_off_t offset, size_t len, WTI_LIVE_RESTORE_HOLE_NODE **holep)
+  wt_off_t offset, size_t len, wt_off_t *hole_begin_off)
 {
-    WTI_LIVE_RESTORE_HOLE_NODE *hole;
-    wt_off_t read_end;
-    bool read_begins_in_hole, read_ends_in_hole;
-
-    if (lr_fh->destination.complete || lr_fh->source == NULL)
-        goto done;
-
-    WT_ASSERT_ALWAYS(session, __wt_rwlock_islocked(session, &lr_fh->ext_lock),
+    /*
+     * The read will be serviced out of the destination if the read is beyond the length of the
+     * source file. This doesn't however account for truncate which could have shortened the
+     * destination to less than that of the source. However the truncate will have marked the
```

Comment:
I'm not sure why we need this sentence. Why do we need to point out truncate shortening the destination file? Here we're only checking the offsets against the source file?

Reply:
That's fair e4729f959